docs/learn/prompts.md

```diff
@@ -1,219 +1,425 @@
 # Prompts
 
-When working with LLMs, a surprising amount of work can go into determining the final text which gets passed into the model. Let's take a look at some of the tools Mirascope provides to make formatting your prompts as seamless as possible.
+Prompts are the foundation of effective communication with Large Language Models (LLMs). Mirascope provides powerful tools to help you create, manage, and optimize your prompts for various LLM interactions. This guide will walk you through the features and best practices for prompt engineering using Mirascope.
 
-## BasePrompt
+## Prompt Templates
 
-`BasePrompt` is a base class which works across all model providers. To create a prompt using `BasePrompt`:
+The primary means of writing prompts in Mirascope is through prompt templates, which are just formatted strings (as they should be) with a few additional conveniences. This allows you to define prompts such that they are dynamic and reusable.
 
-- Define a class which inherits `BasePrompt`
-- Set the prompt via the docstring and denote any template variable with curly braces
-- Type your template variables (and type them correctly! `BasePrompt` is built on Pydantic's `BaseModel` so incorrect typing will result in a `ValidationError`).
+For the purposes of explaining how prompt templates work, we will use:
+
+- `@prompt_template`: a decorator for attaching a prompt template to a function or class
+- `BasePrompt`: a base class for writing provider-agnostic prompts.
+
+We will cover `BasePrompt` in more detail later. For now, all you need to know is that we inject fields with names that match template variables, and we've implemented the `__str__` method for easily viewing how template variables get injected:
```

Comment:
sentence sounds a bit awkward because it's not parallel

Reply:
What do you mean by not parallel?

Reply:
Oh I see it's the "that"